### PR TITLE
Add rule to check for S3 bucket default encryption

### PR DESCRIPTION
**Pull request by msambol**
This rule checks S3 buckets for default encryption.

More on default encryption [here](http://docs.aws.amazon.com/AmazonS3/latest/dev/bucket-encryption.html).

### CODE REVIEW

**Comment by msambol** [comment]
@aaronkhoo Can you review please?

File: python/s3_bucket_default_encryption_enabled.py [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+#            "Effect": "Allow",
+#            "Action": [
+#                "config:PutEvaluations"
+#            ],
+#            "Resource": "*"
+#        },
+#        {
+#            "Effect": "Allow",
+#            "Action": [
+#                "s3:Get*"
```
Comment (kuentzm) [inline python/s3_bucket_default_encryption_enabled.py]: Action is to permissive.  Can it be scoped down to "s3:GetEncryptionConfiguration"

File: python/s3_bucket_default_encryption_enabled.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+#            ],
+#            "Resource": "*"
```
Comment (kuentzm) [inline python/s3_bucket_default_encryption_enabled.py]: Can resource be scoped down to "arn:aws:s3:::*"

**Comment by msambol** [comment]
@kuentzm Fixed. Thanks!

**Comment by msambol** [comment]
@kuentzm Think this is ready. Can you take another look?

**Comment by kuentzm** [comment]
Yes, thanks!  @aaronkhoo will need to review

**Comment by msambol** [comment]
@aaronkhoo this is ready for your review.

**Pull request merged**